rectools/models/nn/transformers/data_preparator.py

```diff
@@ -381,6 +400,17 @@ def transform_dataset_u2i(self, dataset: Dataset, users: ExternalIds) -> Dataset
         # Prepare new user id mapping
         rec_user_id_map = IdMap.from_values(interactions[Columns.User])
 
+        if context is not None:
+            if not (pd.Series(users).isin(context[Columns.User])).all():
+                raise ValueError("No context for all target users")
+            if context.duplicated(subset=Columns.User).any():
+                warnings.warn("Only the earliest row per user is used as context", UserWarning)
```

Comment:
I'd suggest to raise an error here because user may expect completely different result but by some reason ignore warnings